text/0054-rth-payment-id-memos.md

```diff
@@ -0,0 +1,127 @@
+| 24 - 32    | Big-endian bytes of the total outlay amount, as an unsigned 64-bit number |
+| 32 - 40    | Big-endian bytes of 8-byte payment request id number |
+| 40 - 64    | Unused bytes                  |
+
+This is exactly identical to RTH's destination memo except it contains an 8-byte payment request ID number. A user fulfilling a payment request will use this memo and populate that field with the same ID they put into an authenticated sender with payment request ID memo. This will allow the fulfilling user to keep a record that the payment request was fulfilled.
+
+## 0x0102 Authenticated Sender With Payment Id Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 16     | Sender's Address Hash |
+| 16 - 24    | Big-endian bytes of 8-byte payment intent id number |
+| 24 - 48    | Unused bytes          |
+| 48 - 64    | HMAC                  |
+
+This is exactly identical to RTH's authenticated sender with payment request ID memo. The only difference is the memo type. Utilizing the 2-byte memo type, client applications can differentiate payment requests from payment intents.
+
+## 0x0204 Destination With Payment Id Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 16     | Recipient's Address Hash |
+| 16 - 17    | The number of recipients, as an unsigned 8-bit number |
+| 17 - 24    | Big-endian bytes of fee amount, as an unsigned 56-bit number |
+| 24 - 32    | Big-endian bytes of the total outlay amount, as an unsigned 64-bit number |
+| 32 - 40    | Big-endian bytes of 8-byte payment intent id number |
+| 40 - 64    | Unused bytes                  |
+
+This is exactly identical to the herein proposed destination with payment request ID memo. The only difference is the memo type. Utilizing the 2-byte memo type, client applications can differentiate payment requests from payment intents.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There is a limited number of memo types (65536). It likely won't be feasible to add more in the future. This proposal will utilize three more memo types that we won't be able to use for something else in the future.
+
+Additionally, adding three new memo types adds complexity to recoverable transaction history. Many client applications may not decide to utilize payment requests or intents but may wish to implement basic recoverable transaction history. Fortunately, this proposal cements the idea that the sender and destination memos are extended by the payment request ID and payment intent ID variations. It will take additional dev work, but MobileCoin clients and SDKs will be able to use this hierarchy to allow handling of the new memo types the same as the old ones. End-user facing applications may strictly write code to handle the basic sender and destination memo types. This code will always work regardless of any new features that they do not want being added in the future.
```

Comment:
Nit: Capitalize RTH
```suggestion
Additionally, adding three new memo types adds complexity to Recoverable Transaction History. Many client applications may not decide to utilize payment requests or intents but may wish to implement basic recoverable transaction history. Fortunately, this proposal builds on the idea that the sender and destination memos are extended by the payment request ID and payment intent ID variations. It will take additional dev work, but MobileCoin clients and SDKs will be able to use this hierarchy to allow handling of the new memo types the same as the old ones. End-user facing applications may strictly write code to handle the basic sender and destination memo types. This code will always work regardless of any new features that they do not want being added in the future.
```